app/models/user.rb

```diff
@@ -1,12 +1,27 @@
 class User < ActiveRecord::Base
-  validates :Name, presence: true
-  validates :PostsCounter, numericality: { greater_than_or_equal_to: 0 }
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
 
-  def three_recent_posts
-    posts.limit(3).order(created_at: :desc)
-  end
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```

Comment:
- [ ] The application should ask for confirmation of email, you can see that confirmation on rails logs and click on link. To add confirmation of email you have to add  `# :confirmable` to device